File: src/GovUk.Education.ExploreEducationStatistics.Content.Model/Feedback.cs
#nullable enable
using GovUk.Education.ExploreEducationStatistics.Common;
using GovUk.Education.ExploreEducationStatistics.Common.Model;
using System;

namespace GovUk.Education.ExploreEducationStatistics.Content.Model;

public record Feedback : ICreatedTimestamp<DateTime>
{
    public Guid Id { get; set; }

    public DateTime Created { get; set; }

    public required string Url { get; set; }

    public string? UserAgent { get; set; }

    public FeedbackResponse Response { get; set; }

    public string? Context { get; set; }

    public string? Issue { get; set; }

    public string? Intent { get; set; }

    public bool Read { get; set; }
}

File: src/explore-education-statistics-common/src/services/types/feedback.ts
export interface FeedbackRequest {
  url: string;
  userAgent: string;
  response: FeedbackResponse;
  context?: string;
  issue?: string;
  intent?: string;
}

export interface FeedbackViewModel {
  id: string;
  created: Date;
  url: string;
  userAgent: string;
  response: FeedbackResponse;
  context: string;
  intent: string;
  issue: string;
  read: boolean;
}

export type FeedbackResponse = 'Useful' | 'NotUseful' | 'ProblemEncountered';
